### PR TITLE
Source changes towards deep clustering recipe

### PR DESCRIPTION
**Pull request by mpariente**
#### Source changes
- Energy-based VAD which correctly computes VAD independently on batch dim (and any leading dimension)
- Normalize final cost based on VAD mask, independently on batch_dim
- Improve docs in `deep_clustering_loss`. 

#### Egs changes
- Start run.sh with steps and parameters
- Use WHAM-like wsj0-mix preprocessing
- Use WHAM-like data.Dataset (recipe-local for now) 

I'll push the changes to the training and eval script when I have final results, and make another PR for those.

### CODE REVIEW

File: egs/wsj0-mix/DeepClustering/simple_train.py [hunk 1]
```diff
@@ -0,0 +1,238 @@
+import os
```
Comment (sunits) [inline egs/wsj0-mix/DeepClustering/simple_train.py:1]: You can remove this file, if the rest of it works

File: asteroid/losses/cluster.py [hunk 2]
```diff
@@ -1,52 +1,76 @@
...
+    """ Normalize a matrix according to `norm_order`
 
+    Args:
+        matrix (torch.Tensor): Expected shape [batch, *]
+        norm_order (int): Norm order.
+
+    Returns:
+        torch.Tensor, normed matrix of shape [batch]
+    """
+    keep_batch = list(range(1, matrix.ndim))
```
Comment (sunits) [inline asteroid/losses/cluster.py:75]: This was the issue?

Comment (mpariente) [reply asteroid/losses/cluster.py:75]: No this was the not the issue. 
I just made it more general. Before it only worked for 3D tensors. 
For now it is here in the deep clustering loss but we might move it.

File: egs/wsj0-mix/DeepClustering/simple_train.py (hunk 1 above)
Comment (mpariente) [reply egs/wsj0-mix/DeepClustering/simple_train.py:1]: I keep it to have it in the master history but I'll remove it in the recipes's PR.

**Comment by mpariente** [comment]
Thanks for the review @sunits !

**Pull request merged**